crates/router/src/core/payment_methods/vault.rs

```diff
@@ -551,3 +556,80 @@ pub async fn delete_tokenized_data(
             .attach_printable(format!("Got 4xx from the basilisk locker: {err:?}")),
     }
 }
+
+// ********************************************** PROCESS TRACKER **********************************************
+#[cfg(feature = "basilisk")]
+pub async fn add_tokenize_data_task(
+    db: &dyn db::StorageInterface,
+    lookup_key: &str,
+    runner: &str,
+) -> RouterResult<storage::ProcessTracker> {
+    let task = "DELETE_TOKENIZE_DATA";
+    let current_time = common_utils::date_time::now();
+    let tracking_data = serde_json::to_value(api::DeleteTokenizeByTokenRequest {
+        lookup_key: lookup_key.to_owned(),
+    })
+    .into_report()
+    .change_context(errors::ApiErrorResponse::InternalServerError)
+    .attach_printable_lazy(|| format!("unable to convert into value {:?}", lookup_key))?;
+
+    let process_tracker_entry = storage::ProcessTrackerNew {
+        id: format!("{}_{}_{}", runner, task, lookup_key),
+        name: Some(String::from(task)),
+        tag: vec![String::from("BASILISK-V3")],
+        runner: Some(String::from(runner)),
+        retry_count: 0,
+        schedule_time: Some(common_utils::date_time::now() + time::Duration::DAY), // should we keep it in redis, the duration?
```

Comment:
For each data the time of deletion might be different right? the user or application needs to send these then?.

Reply:
should we have redis config as per payment method data ?

Reply:
we need to think about this is based on `fulfillment window` or separate config, if fulfillment window then merchant_account already has it.

For now we proceed with a `default` `redis config` for each `payment_method` and in future will think about fulfillment window.

Reply:
should this be `country` and `payment_method` based ? can't be merchant specific